src/Sema.zig

```diff
@@ -1460,6 +1462,65 @@ fn zirErrorValue(sema: *Sema, block: *Scope.Block, inst: zir.Inst.Index) InnerEr
     });
 }
 
+fn zirErrorToInt(sema: *Sema, block: *Scope.Block, inst: zir.Inst.Index) InnerError!*Inst {
+    const tracy = trace(@src());
+    defer tracy.end();
+
+    const inst_data = sema.code.instructions.items(.data)[inst].un_node;
+    const src = inst_data.src();
+    const operand_src: LazySrcLoc = .{ .node_offset_builtin_call_arg0 = inst_data.src_node };
+    const op = try sema.resolveInst(inst_data.operand);
+    const op_coerced = try sema.coerce(block, Type.initTag(.anyerror), op, operand_src);
+
+    if (op_coerced.value()) |val| {
+        const payload = try sema.arena.create(Value.Payload.U64);
+        payload.* = .{
+            .base = .{ .tag = .int_u64 },
+            .data = (try sema.mod.getErrorValue(val.castTag(.@"error").?.data.name)).value,
+        };
+        return sema.mod.constInst(sema.arena, src, .{
+            .ty = Type.initTag(.u16),
+            .val = Value.initPayload(&payload.base),
+        });
+    }
+
+    try sema.requireRuntimeBlock(block, src);
+    return block.addUnOp(src, Type.initTag(.u16), .error_to_int, op_coerced);
+}
+
+fn zirIntToError(sema: *Sema, block: *Scope.Block, inst: zir.Inst.Index) InnerError!*Inst {
+    const tracy = trace(@src());
+    defer tracy.end();
+
+    const inst_data = sema.code.instructions.items(.data)[inst].un_node;
+    const src = inst_data.src();
+    const operand_src: LazySrcLoc = .{ .node_offset_builtin_call_arg0 = inst_data.src_node };
+
+    const op = try sema.resolveInst(inst_data.operand);
+
+    if (op.value() != null) {
+        const int = try sema.resolveInt(block, operand_src, inst_data.operand, Type.initTag(.u16));
+        if (int > sema.mod.global_error_set.count() or int == 0)
+            return sema.mod.fail(&block.base, operand_src, "integer value {d} represents no error", .{int});
+        const payload = try sema.arena.create(Value.Payload.Error);
+        payload.* = .{
+            .base = .{ .tag = .@"error" },
+            .data = .{ .name = sema.mod.inv_error_map.items[int - 1] },
```

Comment:
Ah I see, so there's a -1 here to account for the array list being off by one. Can I suggest a slightly modified strategy here? In stage1 we just add a 0 => "(no error)" mapping on startup to this array before analyzing any code. I think that's the way to go here. Then the array list mapping will be matched up exactly, no off by one errors to trip over.